src/jobs/taskcluster_graph.js

```diff
@@ -29,6 +36,32 @@ function parseUrl(url) {
   };
 }
 
+/**
+Fetch a task graph from a url (retires included...)
+*/
+async function fetchGraph(job, url) {
+  assert(url, 'url is required');
+  job.log(`fetching graph ${url}`);
+  let currentRetry = 0;
+  while (currentRetry++ < GRAPH_RETIRES) {
+    try {
```

Comment:
This loop is a great candidate for a generic retry() function. Or some existing module like https://github.com/FGRibreau/node-request-retry. I'll be much easier to read, and I bet this pattern will be used somewhere else.